official/resnet/resnet_model.py

```diff
@@ -205,162 +205,126 @@ def projection_shortcut(inputs):
         data_format=data_format)
 
   # Only the first block per block_layer uses projection_shortcut and strides
-  inputs = block_fn(inputs, filters, is_training, projection_shortcut, strides,
+  inputs = block_fn(inputs, filters, training, projection_shortcut, strides,
                     data_format)
 
   for _ in range(1, blocks):
-    inputs = block_fn(inputs, filters, is_training, None, 1, data_format)
+    inputs = block_fn(inputs, filters, training, None, 1, data_format)
 
   return tf.identity(inputs, name)
 
 
-def cifar10_resnet_v2_generator(resnet_size, num_classes, data_format=None):
-  """Generator for CIFAR-10 ResNet v2 models.
-
-  Args:
-    resnet_size: A single integer for the size of the ResNet model.
-    num_classes: The number of possible classes for image classification.
-    data_format: The input format ('channels_last', 'channels_first', or None).
-      If set to None, the format is dependent on whether a GPU is available.
-
-  Returns:
-    The model function that takes in `inputs` and `is_training` and
-    returns the output tensor of the ResNet model.
-
-  Raises:
-    ValueError: If `resnet_size` is invalid.
+class Model(object):
+  """Base class for building the Resnet v2 Model.
   """
-  if resnet_size % 6 != 2:
-    raise ValueError('resnet_size must be 6n + 2:', resnet_size)
+  resnet_size = None
+  num_classes = None
+  num_filters = None
+  kernel_size = None
+  first_pool_size = None
+  second_pool_size = None
+  block_fn = None
+  layers = None
+  stride_sizes = None
+  final_size = None
+  data_format = None
+  input_shape = None
+
+  def __init__(self, resnet_size, num_classes, num_filters, kernel_size,
+               first_pool_size, second_pool_size, block_fn, layers,
+               stride_sizes, final_size, data_format=None):
+    """Creates a model for classifying an image.
+
+    Args:
+      resnet_size: A single integer for the size of the ResNet model.
+      num_classes: The number of classes used as labels.
+      num_filters: The number of filters to use for the model.
+      kernel_size: The kernel size to use for convolution.
+      first_pool_size: Pool size to be used for the first pooling layer.
+        If none, the first pooling layer is skipped.
+      second_pool_size: Pool size to be used for the second pooling layer.
+      block_fn: Which block layer function should be used? Pass in one of
+        the two functions defined above: building_block or bottleneck_block
+      layers: A list containing n values, where n is the bnumber of block
```

Comment:
`layers` seems like a very ambiguous name (especially since there are other "layers" like the pooling ones and the final layer that aren't affected by this). How about we call out all the "block" affecting parameters together. So `layers` becomes `block_sizes` or `num_blocks` and `stride_sizes` becomes `block_strides` (see comment below).

Then we have three "block" parameters: `block_fn`, `block_sizes`, `block_strides`.